### DISCUSSION TITLE
It is possible to override ` --install-completion` option ?

### DISCUSSION THREAD

**Discussion by Fasamik39** [fastapi/typer#564] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# I want to get something like this

def install_completion_option():
    answer = input("Are you sure you want to use the --install-completion command?"
               " This will make changes to the .bashrc file")
    if answer == "yes":
        # --install-completion is performed here
```

### Description

I want to override the functionality of the `--install-completion` command. I want to warn the user that the `.bashrc` file will be modified after this operation. Is it possible ?

I am trying to run the `install()` function from the `typer.completion` module as follows:

```python
import click
import typer
from typer.completion import install
from typer.core import TyperGroup as TyperGroupBase

app = typer.Typer(
    name="capsule", help="capsule CLI", no_args_is_help=True, cls=TyperGroup, add_completion=False
)

@app.callback(
    invoke_without_command=True,
    help="""capsule: a packaging system for notebook-based projects.""",
)
def main(
    install_completion: Optional[bool] = typer.Option(
        False, "--install-completion", help="Install completion"
    ),
) -> None:
    if install_completion:
        answer = input(
            "Are you sure you want to use the --install-completion command?"
            " This will make changes to the .bashrc file")
        if answer == "yes":
            install()
    typer.Exit()

if __name__ == "__main__":
    app()

```

but I didn't get what I want because when I press TAB it always shows "plain" before the option is completed like this

```
myapp n[TAB] -> myapp plain, new
```
instead:

```
myapp n[TAB] -> myapp new
```

### Operating System

Linux

### Operating System Details

Ubuntu 22.04.2 LTS

### Typer Version

0.7.0

### Python Version

Python 3.11.0

**Comment by AngryMaciek** [comment]
I would also be interested in a solution to this one...

**Comment by FilipSchymik** [comment]
Interesting ⏫

**Comment by vesmanojlovic** [comment]
Bump ⏫
